tensorboard/plugins/graph/tf_graph_common/tf-graph-icon.html

```diff
@@ -0,0 +1,251 @@
+            fill$="[[_fill]]"
+            stroke$="[[_stroke]]"
+            class$="{{_fadedClass(faded, 'rect')}}"
+            stroke-width="2px"
+            height="14" width="35"
+            rx="5" ry="5"/>
+      </svg>
+    </template>
+    <template is="dom-if" if="[[_isType(type, 'SERIES')]]">
+      <template is="dom-if" if="[[vertical]]">
+        <svg height$="[[height]]"
+            preserveAspectRatio="xMinYMid meet" viewBox="0 0 16 15">
+          <use xmlns:xlink="http://www.w3.org/1999/xlink"
+              xlink:href="#op-series-vertical-stamp"
+              fill$="[[_fill]]"
+              stroke$="[[_stroke]]"
+              class$="{{_fadedClass(faded, 'series')}}"
+              x="0" y="2" />
+        </svg>
+      </template>
+      <template is="dom-if" if="[[!vertical]]">
+        <svg height$="[[height]]"
+            preserveAspectRatio="xMinYMid meet" viewBox="0 0 24 10">
+          <use xmlns:xlink="http://www.w3.org/1999/xlink"
+              xlink:href="#op-series-horizontal-stamp"
+              fill$="[[_fill]]"
+              stroke$="[[_stroke]]"
+              class$="{{_fadedClass(faded, 'series')}}"
+              x="0" y="1" />
+        </svg>
+      </template>
+    </template>
+  </template>
+
+  <script>
+    (function() {
+      const GRAPH_ICON_TYPE = {
+        CONST: 'CONST',
+        META: 'META',
+        OP: 'OP',
+        SERIES: 'SERIES',
+        SUMMARY: 'SUMMARY',
+      };
+
+      Polymer({
+        is: 'tf-graph-icon',
+
+        properties: {
+          /**
+           * Type of node to draw.
+           * @param {GRAPH_ICON_TYPE}
+           */
+          type: String,
+
+          /** Direction for series. */
+          vertical: {
+            type: Boolean,
+            value: false,
+          },
+
+          /**
+           * Fill for the icon, optional. If fill is specified and node is not
+           * specified, then this value will override the default for the
+           * element. However, if node is specified, this value will be ignored.
+           */
+          fillOverride: {
+            type: String,
+            value: null,
+          },
+
+          strokeOverride: {
+            type: String,
+            value: null,
+          },
+
+          /** Height of the SVG element in pixels, used for scaling. */
+          height: {
+            type: Number,
+            value: 20,
+          },
+
+          faded: {
+            type: Boolean,
+            value: false,
+          },
+
+          _fill: {
+            type: String,
+            computed: '_computeFill(type, fillOverride)',
+          },
+
+          _stroke: {
+            type: String,
+            computed: '_computeStroke(type, strokeOverride)',
+          },
+        },
+
+        getSvgDefineableElement() {
+          return this.$.svgDefs;
+        },
+
+        _computeFill(type, fillOverride) {
+          if (fillOverride != null) return fillOverride;
+          switch (type) {
```

Comment:
Discussed offline: We agree that TypeScript is the desired end state for
these files, and `graph-icon` is easy enough to convert, but we’re
leaving `node-icon` as is because it would require more substantive
changes (fixing type errors, etc.) than we’d prefer to make in a “just
get Polymer 2 working” commit.